test/system_category_test.cpp

```diff
@@ -102,6 +102,7 @@ static void test_message( sys::error_category const & cat, int ev )
 
     char buffer[ 4096 ]; // yes, really
     BOOST_TEST_CSTR_EQ( cat.message( ev, buffer, sizeof( buffer ) ), sys_strerror( ev ).c_str() );
+    BOOST_TEST_CSTR_EQ( cat.message( ev ).c_str(), buffer );
```

Comment:
I can revert this if you see it is unneeded, but I find this untested.

Reply:
It doesn't hurt I suppose.

Reply:
Yeah, it does hurt. :-)

Reply:
`.message()` doesn't necessarily fill `buffer`, it's allowed to return a pointer to the message elsewhere.

Reply:
Funny enough I have 'optimized' it before opening the PR from `BOOST_TEST_EQ( cat.message( ev ), sys_strerror( ev ) );`. How do I proceed?

Reply:
Well this addition (line 105) needs to go, it causes all Travis tests to fail.